### PR TITLE
Changed client structure

### PR DESCRIPTION
**Pull request by jBenderToledo**
Untested. Intent is to make each message check hidden by the API for maintainability reasons.

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -8,32 +8,48 @@
     data = json.load(json_file)
     TOKEN = data['discordKey']
 
-client = discord.Client()
+client = commands.Bot(command_prefix = '!')
+
+#####################
```
Comment (joey-kilgore) [inline bot.py]: I think we need to discuss our coding style because I'm not sure how much we will want these big block comments. Especially because every method has the ```@client.event``` or ```@client.command``` before it.

Comment (jBenderToledo) [reply bot.py]: Yes, we should. What sort of standard should we go by? I figured that it would make it easier to read -- sorry if that was a bit silly.

File: terminal.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import subprocess
+import os
+from pathlib import Path
+
+# Class to represent an instance of a terminal which is initiated
```
Comment (joey-kilgore) [inline terminal.py]: In bot.py we have the method/class headers on the lines following the method/class, but here they are before. We need to pick one.

File: terminal.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        # Object containing data returned by the terminal after the command was executed
+        return ({
+            'terminalName': self.name,
+            'currentDirectory': self.currentDirectory,
+            'executedCommand': command,
+            'outputString': stdout.decode('utf-8'),
+            'errorString': stderr.decode('utf-8')
+        })
+
+# ~~~~~~~~~~~~~~~~ TEST CODE (for windows) ~~~~~~~~~~~~~~~~
```
Comment (joey-kilgore) [inline terminal.py]: The current state has not been tested for windows, can we test the changing directory with the optional currentDirectory parameter. @Jacob-Hunt

File: bot.py [hunk 4]
```diff
@@ -2,38 +2,213 @@
...
+@client.command(name='shutdown')
+async def shutdown(ctx):
+    # Clarifies to the user that the bot is shutting down. Logs out. Tells the host machine that the bot has logged out.
+    await ctx.send("Shutting down...")
+    logger.log("{0.user.name} has logged out.".format(client))
+    await client.logout()
+
+
+	
+# The following segment of code was written by @TheToiletTyrant (316thebrains)
```
Comment (joey-kilgore) [inline bot.py]: This code is irrelevant since we have git commits with authors. Unless we are programming before source control was created this isn't necessary

File: bot.py [hunk 5]
```diff
@@ -2,38 +2,213 @@
...
+
+#############################################################
+# This code has been QUARENTINED due to it's infinite loops #
```
Comment (joey-kilgore) [inline bot.py]: Should we remove this then for the first release, or should we try and fix it. It seems extremely different then the rest of the commands (using sockets to make a ping) and maybe should be part of a later release with some sort of web scraping functionality?

**Review by joey-kilgore — COMMENTED** [review]
Overall the functionality is there, but this needs to be cleaned up before the first release.

File: terminal.py (hunk 3 above)
Comment (Jacob-Hunt) [reply terminal.py]: @joey-kilgore Can you rephrase this comment? What, specifically, do you want tested for windows? Thanks!

Comment (joey-kilgore) [reply terminal.py]: There is an optional third parameter for setting the current directory when creating a terminal instance. We should test this and ensure you can have different terminal directories called when instantiating bother the Terminal class and the Popen class

Comment (Jacob-Hunt) [reply terminal.py]: @joey-kilgore Got it! Will test this on my Windows machine and report back shortly...

Comment (Jacob-Hunt) [reply terminal.py]: @joey-kilgore Looks like we're good:

terminal-py_execution

File: terminal.py (hunk 2 above)
Comment (Jacob-Hunt) [reply terminal.py]: Added to to-do list for 8/23

File: bot.py (hunk 5 above)
Comment (Jacob-Hunt) [reply bot.py]: Maybe clear it for the first release and create a "suggested features" issue-page for it?

**Review by joey-kilgore — COMMENTED** [review]
@Jacob-Hunt if you delete the .suo and .sqlite files (presumably generated when someone opened this in visual studo) we can finish this.

**Review by joey-kilgore — APPROVED** [review]
